rust/frontend/src/server.rs

```diff
@@ -254,25 +271,50 @@ impl FrontendServer {
 // These handlers simply proxy the call and the relevant inputs into
 // the appropriate method on the `FrontendServer` struct.
 
+/// Health check endpoint that returns 200 if the server and executor are ready
+#[utoipa::path(
+    get,
+    path = "/api/v2/healthcheck",
+    responses(
+        (status = 200, description = "Success", body = String, content_type = "application/json"),
+        (status = 503, description = "Service Unavailable", body = ErrorResponse),
+    )
+)]
 async fn healthcheck(State(server): State<FrontendServer>) -> impl IntoResponse {
     server.metrics.healthcheck.add(1, &[]);
     let res = server.frontend.healthcheck().await;
     let code = match res.get_status_code() {
         tonic::Code::Ok => StatusCode::OK,
         _ => StatusCode::SERVICE_UNAVAILABLE,
     };
-
     (code, Json(res))
 }
 
+/// Heartbeat endpoint that returns a nanosecond timestamp of the current time. Useful for making sure the client remains connected.
+#[utoipa::path(
+    get,
+    path = "/api/v2/heartbeat",
+    responses(
+        (status = 200, description = "Success", body = HeartbeatResponse),
+        (status = 500, description = "Server error", body = ErrorResponse)
+    )
```

Comment:
is there a way to define a common set of error types instead of re-defining for each route?

Reply:
Good feedback - turns out there is. Working on implementing this now.

Reply:
https://docs.rs/utoipa-gen/latest/utoipa_gen/derive.IntoResponses.html

Reply:
I looked into the IntoResponses, and it seems that we can't mix per-path response (like HeartBeatResponse) with the CommonResponses. So, it won't work for our use case